ecleankernel/layout/blspec.py

```diff
@@ -45,19 +45,27 @@ def __init__(self,
         for path in ("etc/kernel/entry-token", "etc/machine-id"):
             try:
                 with open(root / path) as f:
-                    kernel_id = f.read().strip()
+                    self.kernel_id = f.read().strip()
                 break
             except FileNotFoundError:
                 pass
         else:
             raise LayoutNotFound("/etc/machine-id not found")
 
-        for d in self.potential_dirs:
-            self.bootdir = root / d / kernel_id
-            if self.bootdir.is_dir():
-                return
-        else:
-            raise LayoutNotFound(f"/boot/[EFI/]{kernel_id} not found")
+        bootctl = subprocess.check_output(["bootctl", "-x"], text=True)
```

Comment:
I'm not convinced I'm fine with adding a dependency on systemd here. Even though I suppose it's unlikely people would still use gummiboot, it's a heavy dependency to add to the ebuild.

Reply:
AFAIK `systemd-utils[boot]` also provides the bootctl command, so no need to depend on systemd.

Reply:
> AFAIK `systemd-utils[boot]` also provides the bootctl command, so no need to depend on systemd.

Double checked and I can confirm that bootctl is also available on non-systemd systems via `systemd-utils[boot]`. So users of systemd-boot (regardless of init system) should already have it available, no need to add systemd as a hard requirement.

I strongly prefer this or something similar over guessing where BOOT or ESP is.

Reply:
Another option is to revert to the `potential_dirs` stuff, but check for the directory that contains the `loader/loader.conf`. 

It used to check for the directory that contains the directory `$ENTRY_TOKEN`, but this will not work on a system that contains only kernels installed using the type 2 layout since then this directory won't exist. However, `loader/loader.conf` (and the other files in `loader/`) will always exist since they are created when running `bootctl install`.

I really think asking bootctl is the better solution though, it seems to me more robust.

Reply:
I'd rather stay conservative here. `/boot` and kernels (blspec or not) are very fragile and subject to a lot of random customization, and every change is likely to break something for someone.